### PR TITLE
guide: remove refences to outdated secondary checkers

### PR DESCRIPTION
**Pull request by ordian**
Previously backing/approval was called primary/secondary checks. It was later renamed, because approval voters provide most of the shared security, whereas "secondary" does not convey this meaning fully. 

I've also removed "Router" module from the glossary since I believe this is obsolete by XCM router. @pepyakin LMK if that's correct.

Added "SessionInfo" and "Slashing" modules to the runtime/README.md.

### CODE REVIEW

**Review by pepyakin — COMMENTED** [review]
I think when the Router entry was added I think it was pretty much not a thing. The reason it was added is to make sure there is a searchable entry for people who is familiar with the old name but not the new name.

But I think it hardly matters. It's not like we have a lot of materials mentioning "secondary" checkers around.

**Comment by ordian** [comment]
> I think when the Router entry was added I think it was pretty much not a thing. The reason it was added is to make sure there is a searchable entry for people who is familiar with the old name but not the new name.

I see, it was introduced in #1939 when the module was disassembled into ump, dmp, hrmp.

> But I think it hardly matters. It's not like we have a lot of materials mentioning "secondary" checkers around.

Do think you I should close the PR? I believe having multiple terms for the same thing adds confusion for new people lacking the historical context.

**Comment by pepyakin** [comment]
It's inconsequential and I don't care that much.

Good point about confusingness. It could be mitigated by either an entry for the old definition with the reference ("Deprecated. See X", where X is the new term) or updating the new definition with "fka Y" where Y is the old term.

File: roadmap/implementers-guide/src/glossary.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 Here you can find definitions of a bunch of jargon, usually specific to the Polkadot project.
 
+- **Approval Checker:** A validator who has been randomly selected to perform validity checks on a parablock which is pending approval.
```
Comment (burdges) [inline roadmap/implementers-guide/src/glossary.md:5]: "who randomly self-selects so" maybe?

**Comment by burdges** [comment]
We sometimes explain that "fishermen" were removed because they were always fictitious economically, both when we originally proposed them and when others do under the name "watch towers".  We do not need to talk about this in the guide however, so nothing to change there.

If you do keep "router" then maybe say "message router"?  I've no opinion but "router" by itself feels confusing.

**Comment by ordian** [comment]
> We sometimes explain that "fishermen" were removed because they were always fictitious economically, both when we originally proposed them and when others do under the name "watch towers". We do not need to talk about this in the guide however, so nothing to change there.

Thanks for the reminder about the old references to "fishermen". I've removed them in [2716b2a](https://github.com/paritytech/polkadot/pull/6309/commits/2716b2abe352985d3314eeb44ace6b573ba8f9ad).

There's also an outdated doc https://github.com/paritytech/polkadot/blob/master/roadmap/parachains.md, but it's quite outdated in other areas as well and not linked from the book.

> It could be mitigated by either an entry for the old definition with the reference ("Deprecated. See X", where X is the new term) or updating the new definition with "fka Y" where Y is the old term.

I feel like for the curious software archaeologists, there are tools like git blame. 

But I'm also curious about what new joiners think cc @m-cat @alexgparity @BradleyOlson64

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2060828

**Comment by burdges** [comment]
[2716b2a](https://github.com/paritytech/polkadot/pull/6309/commits/2716b2abe352985d3314eeb44ace6b573ba8f9ad) looks good.  There was an intermediate state where a fishermen could be basically anybody, like they'd just pay for escalation so that everyone checks, but you've cleaned up the mess nicely.

**Comment by alexgparity** [comment]
> I feel like for the curious software archaeologists, there are tools like git blame.

I agree. I'd rather have a guide focused on the current system. And maybe as a single source of truth? It would make onboarding easier to not have to read/understand about obsolete ideas/approaches and not have to jump back and forth between the guide and blog posts.

**Comment by m-cat** [comment]
I agree with @alexgparity, I would prefer the guide be kept up-to-date.

**Comment by BradleyOlson64** [comment]
I'm with @alexgparity and @m-cat. Clarity about the current design is paramount.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2068264

**Review by m-cat — APPROVED** [review]
> Previously backing/approval was called primary/secondary checks. It was later renamed, because approval voters provide most of the shared security, whereas "secondary" does not convey this meaning fully.

Makes sense and agreed, thanks for the background.

**Review by sandreim — APPROVED** [review]

**Pull request merged**